### ISSUE TITLE
StartFrameReaderAsync can't be stopped when rtsp is offline

### ISSUE THREAD

**Issue by JPasterkampRotec** [nager/Nager.VideoStream#9]
I'm using Nager.VideoStream for getting frames from a security camera on a Windows 10 PC with a .NET framework 4.8 application.

When the source is offline/not reachable, it is not possible to stop StartFrameReaderAsync.
Even when you cancel the provided CancellationToken, StartFrameReaderAsync won't stop.
The only way to then stop it, is by closing my entire application.

I reproduced this issue in the Nager.VideoStream.TestConsole by only changing:
- The input source: ` var inputSource = new StreamInputSource("my_secret_rtsp_url");`
- The location of the ffmpeg.exe: `var client = new VideoStreamClient(@"my_secret_folder\ffmpeg.exe");`

The console application clearly cancels the cancellation token:
```
Console.WriteLine("Press any key for stop");
Console.ReadKey();
cancellationTokenSource.Cancel();

Console.WriteLine("Press any key for quit");
Console.ReadKey();
```

But this console writeline is never executed:
```
Console.WriteLine("End Stream Processing");
```

In my application I need to call StartFrameReaderAsync  quite frequently. 
That means without proper cancellation, multiple StartFrameReaderAsync threads will be active at the same time. 
This is a major performance issue for my application when the camera is offline/not reachable.

**Comment by tinohager** [comment]
In the meantime, have you found a solution to the problem?

**Comment by JPasterkampRotec** [comment]
No, but the code keeps hanging at `var length = await frameOutputStream.ReadAsync(buffer, 0, buffer.Length);` when the source is offline.

I read this old ticket from FFMPEG (https://trac.ffmpeg.org/ticket/2294), which seems to suggest that parameter `stimeout` solves this issue. 
**I tried this with the Nager.VideoStream.TestConsole and it always immediately times out, even when the source is available and I put `-stimeout 123456789`  in the `ProcessStartInfo` arguments.**
I hope to have sent you in the right direction with this information.

**Comment by blackholeearth** [comment]
**why it might be hanging forever ?**
A:  default behavior is wait forever. 

https://ffmpeg.org/ffmpeg-protocols.html

> timeout
>    Set timeout in microseconds of socket I/O operations used by the underlying low level operation. By default it is set to -1, which means that the timeout is not specified.

-----------
**how people fixed it?**
A: adding this params to ffmeg.exe    ``-rw_timeout 5000000``      process exits after 5seconds.

 https://stackoverflow.com/questions/71095477/ffmpeg-timeout-with-rtsp

-------------

[ffmpeg.com -> rw_timeout](https://ffmpeg.org/ffmpeg-protocols.html#toc-Protocols:~:text=the%20following%20options%3A-,rw_timeout,-Maximum%20time%20to):
> rw_timeout  
Maximum time to wait for (network) read/write operations to complete, in microseconds.

https://stackoverflow.com/questions/35123267/how-to-set-rtsp-connection-timeout-in-ffmpeg

**Comment by JPasterkampRotec** [comment]
> **why it might be hanging forever ?** A: default behavior is wait forever.
> 
> https://ffmpeg.org/ffmpeg-protocols.html
> 
> > timeout
> > Set timeout in microseconds of socket I/O operations used by the underlying low level operation. By default it is set to -1, which means that the timeout is not specified.
> 
> **how people fixed it?** A: adding this params to ffmeg.exe `-rw_timeout 5000000` process exits after 5seconds.
> 
> https://stackoverflow.com/questions/71095477/ffmpeg-timeout-with-rtsp
> 
> [ffmpeg.com -> rw_timeout](https://ffmpeg.org/ffmpeg-protocols.html#toc-Protocols:~:text=the%20following%20options%3A-,rw_timeout,-Maximum%20time%20to):
> 
> > rw_timeout
> > Maximum time to wait for (network) read/write operations to complete, in microseconds.
> 
> https://stackoverflow.com/questions/35123267/how-to-set-rtsp-connection-timeout-in-ffmpeg

This did not solve the issue sadly. I will create a pull request soon, as I have discovered a way to solve the issue.
